### ISSUE TITLE
[Bug]: CLI --override flag is ignored by the finetuning command

### ISSUE THREAD

**Issue by pranaysb** [NVIDIA/NeMo-Agent-Toolkit#2102]
### Version

0.1.dev

### Which installation method(s) does this occur on?

Source

### Describe the bug.

When running `nat finetune` with the `--override` flag (e.g., `nat finetune --override "finetuning.run_configuration.num_epochs=10"`), the configuration overrides are entirely ignored by the runtime. 

The CLI parser in `nat/cli/commands/finetune.py` correctly captures the flag, but there is a stale comment on line 112: `# TODO: Implement config override logic similar to other commands`. 

As a result, the underlying orchestrator in `finetuning_runtime.py` calls the standard `load_config(run_config.config_file)`, ignoring the captured `run_config.override` tuples entirely.

**Expected behavior:**
The `finetune` command should behave exactly like the `run` or `serve` commands, dynamically injecting CLI overrides into the configuration dictionary prior to Pydantic validation via `load_and_override_config`.

### Minimum reproducible example

```shell
# This command runs, but the override is ignored, and the default epochs from the YAML (or schema) are used instead.
nat finetune --override "finetuning.run_configuration.num_epochs=10"
```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

[No explicit stack trace; the system silently uses the default config values instead of the provided overrides, failing silently in the background config resolution phase.]

</pre></details>

### Other/Misc.

**Possible Solution**:
I have already audited the codebase and locally validated a fix. The solution is to update `packages/nvidia_nat_core/src/nat/finetuning/finetuning_runtime.py` to use `load_and_override_config` alongside `validate_schema`, properly passing `run_config.override`. I will be opening a PR shortly!

### Code of Conduct

- [x] I agree to follow the NeMo Agent Toolkit Code of Conduct
- [x] I have searched the [open bugs](https://github.com/NVIDIA/NeMo-Agent-Toolkit/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

**Comment by mdemoret-nv** [comment]
Thanks for reporting. If you want to open a PR to fix this, that would be greatly appreciated!!

**Comment by pranaysb** [comment]
> Thanks for reporting. If you want to open a PR to fix this, that would be greatly appreciated!!

Thanks for taking a look, @mdemoret-nv! I've actually just gone ahead and opened a PR for this over at #2103. 
Let me know if there's anything you'd like me to change or adjust in the implementation!
